plonky3/src/verifier.rs

```diff
@@ -10,250 +12,423 @@ use p3_matrix::dense::RowMajorMatrixView;
 use p3_matrix::stack::VerticalPair;
 use tracing::instrument;
 
+use crate::circuit_builder::{PowdrCircuit, PowdrTable};
+use crate::params::{Challenge, Challenger, Commitment, Plonky3Field, ProverData};
 use crate::symbolic_builder::{get_log_quotient_degree, SymbolicAirBuilder};
-use crate::{MultiStageAir, Proof, StarkVerifyingKey, VerifierConstraintFolder};
+use crate::{
+    FieldElementMap, MultiStageAir, Proof, StageOpenedValues, StarkVerifyingKey, TableOpenedValues,
+    VerifierConstraintFolder,
+};
 use p3_uni_stark::{PcsError, StarkGenericConfig, Val};
 
-#[instrument(skip_all)]
-pub fn verify<SC, A>(
-    config: &SC,
-    air: &A,
-    challenger: &mut SC::Challenger,
-    proof: &Proof<SC>,
-    public_values: &Vec<Val<SC>>,
-) -> Result<(), VerificationError<PcsError<SC>>>
+/// A sub-table to be proven, in the form of an air and values for the public inputs
+struct Table<'a, T: FieldElementMap>
 where
-    SC: StarkGenericConfig,
-    A: MultiStageAir<SymbolicAirBuilder<Val<SC>>>
-        + for<'a> MultiStageAir<VerifierConstraintFolder<'a, SC>>,
+    ProverData<T>: Send,
+    Commitment<T>: Send,
 {
-    verify_with_key(config, None, air, challenger, proof, vec![public_values])
+    air: PowdrTable<'a, T>,
+    public_values_by_stage: &'a [Vec<Val<T::Config>>],
+}
+
+impl<'a, T: FieldElementMap> Table<'a, T>
+where
+    ProverData<T>: Send,
+    Commitment<T>: Send,
+{
+    fn get_log_quotient_degree(&self) -> usize {
+        get_log_quotient_degree(
+            &self.air,
+            &self
+                .public_values_by_stage
+                .iter()
+                .map(|values| values.len())
+                .collect::<Vec<_>>(),
+        )
+    }
 }
 
 #[instrument(skip_all)]
```

Comment:
what's being skipped here?

Reply:
Good question, this is a leftover from uni-stark